os/binfmt/libelf/libelf_addrenv.c

```diff
@@ -105,7 +105,7 @@ static void allocateregions(FAR struct elf_loadinfo_s *loadinfo)
 		}
 	}
 
-	loadinfo->binp->ramstart = *allocs[0] = (uintptr_t)kumm_memalign(sizes[0], totalsize);
+	loadinfo->binp->ramstart = *allocs[0] = (uintptr_t)kmm_memalign(sizes[0], totalsize);
```

Comment:
It means we will allocate the user binary's section into the kernel side. Is it okay?

Reply:
Yes. We will allocate user sections from kernel heap, and then we will use the mpu to provide permissions as required for each section.

Reply:
In this case, kernel developer should know how big the user binary is. I think it is not easy.

Reply:
In any case there needs to be a balance between the kernel and user areas, and it always depends on the size of the user apps. If we allocate app sections from user heap, then we need to know before hand, the size of user heap. Even in this case, the developer needs to split the ram between user and kernel based on the app size. 
I think this method is better because, in this case we dont have to predetermine the size of the user heap. If the app size is small enough, then there is no problem from kernel side. On the other hand, if the app size becomes too big and takes up too much kernel heap, then having the app allocation from a separate user heap would also cause the same problem.